compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2547,6 +2544,18 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             "ban_nonexisting_field: field={:?}, base={:?}, expr={:?}, base_ty={:?}",
             ident, base, expr, base_ty
         );
+
+        for (ty, _) in self.autoderef(expr.span, base_ty) {
+            if let ty::Adt(def, _) = ty.kind()
+                && !def.is_enum()
+            {
+                let variant = def.non_enum_variant();
+                if let Some(taint) = variant.has_errors() {
+                    return taint;
+                }
+            }
+        }
```

Comment:
Additions like this are what I would like to avoid, as we have to add them piecemeal to fix ICEs. Not sure yet how to systematically fix that